xdsl/backend/riscv/lowering/convert_arith_to_riscv.py

```diff
@@ -93,74 +93,54 @@ def match_and_rewrite(
         )
 
 
-class LowerBinaryOp(RewritePattern):
```

Comment:
this was just too clever by half, two separate classes, one for floats and one for ints is less code than this one with all its helpers